Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteAdapterProxy.h

```diff
@@ -69,21 +69,27 @@ class RemoteAdapterProxy final : public PAL::WebGPU::Adapter {
     template<typename T>
     WARN_UNUSED_RETURN bool send(T&& message)
     {
+        if (m_destroyed)
```

Comment:
I'm not sure if there is a better / alternate way of achieving this, maybe @kkinnunen-apple knows?

It is to avoid the issue where destroy is sent from, for example, GPUAdapter prior to the destructor of GPUAdapter being called. But then destroy gets sent again in the destructor of GPUAdapter, but there is no receiver for the message so we encounter an IPC error.

The logic doesn't really apply to GPUAdapter but does apply to time limited resources, e.g., GPUCommandEncoder, GPUTextureView, etc

Reply:
Can't we avoid this by construction, and say "the only place that sends destroy messages are the destructors of RemoteFooProxy classes?"